test/digraph-cyclicity-hadwick-james.jl

```diff
@@ -0,0 +1,31 @@
+@testset "Hadwick-James Cycles" begin
+    hadwick_james_ex1 = DiGraph(5)
+    add_edge!(hadwick_james_ex1, 1, 2)
+    add_edge!(hadwick_james_ex1, 2, 3)
+    add_edge!(hadwick_james_ex1, 3, 4)
+    add_edge!(hadwick_james_ex1, 3, 5)
+    add_edge!(hadwick_james_ex1, 4, 5)
+    add_edge!(hadwick_james_ex1, 5, 2)
+
+    hadwick_circuits = simplecycles_hadwick_james(hadwick_james_ex1)
+
+    expected_circuits = Vector{Int}[
+        [2, 3, 4, 5],
+        [2, 3, 5]
+    ]
+    @test issubset(expected_circuits, hadwick_circuits)
```

Comment:
Check how we use `make_testdigraphs` in `runtests.jl` to ensure this works on other digraphs with different eltypes.